### ISSUE TITLE
Repository used in the mongodb pom.xml not directly accessible

### ISSUE THREAD

**Issue by gkamat** [brianfrankcooper/YCSB#370]
The pom.xml file for the MongoDB binding contains a reference to the following repository:

  http://www.allanbank.com/repo

However trying to access it directly does not work, for example:

$ wget http://www.allanbank.com/repo/
--2015-07-24 14:50:24--  http://www.allanbank.com/repo/
Resolving www.allanbank.com (www.allanbank.com)... 206.210.70.161
Connecting to www.allanbank.com (www.allanbank.com)|206.210.70.161|:80... connected.
HTTP request sent, awaiting response... 403 Forbidden
2015-07-24 14:50:24 ERROR 403: Forbidden.

Also, it would also be good to use the https transport here, if possible.

**Comment by allanbank** [comment]
Yes, file listing is turned off on the server since it is not needed.  If you try to download the artifacts directly (see links below) it works. More details here:http://www.allanbank.com/mongodb-async-driver/userguide/maven.html

https is on the todo list.

http://www.allanbank.com/repo/com/allanbank/mongodb-async-driver/2.0.1/mongodb-async-driver-2.0.1.jar
http://www.allanbank.com/repo/com/allanbank/mongodb-async-driver/2.0.1/mongodb-async-driver-2.0.1-sources.jar
http://www.allanbank.com/repo/com/allanbank/mongodb-async-driver/2.0.1/mongodb-async-driver-2.0.1.pom

**Comment by mthak** [comment]
The problem we have is "if this artifact changes to something else tomorrow, or if we add more artifacts like this in future" . Without a mirror It would need someone to download the artifacts manually and upload it on our internal maven repository.

**Comment by busbey** [comment]
@mthak isn't that why folks set up nexus (or whatever repository) to mirror things? How is this particular repo different from the other ones we use in e.g. for tarantool or gemfire?

**Comment by gkamat** [comment]
It would be helpful to have maven mirror/snapshot all dependent repositories at each YCSB release, and use those for the build.  Perhaps there is a way to enhance the build specification to do that.

**Comment by busbey** [comment]
I was checking and I believe the maven-dependency-plugin should make this easy enough to do on releases.

I'll make a go at putting one in place for the 0.2.0 Release and we can see what would make it sustainable.

**Comment by risdenk** [comment]
Is the @allanbank repo even necessary any more with https://mvnrepository.com/artifact/org.mongodb/mongodb-driver-async? Looks like there is an async driver directly from MongoDB that is newer?

I'll try to put together a PR to address.

**Comment by manolama** [comment]
Looks like the switch is pretty involved. May bump this to the next release.
